### PR TITLE
feat: Added Vite and tasks

### PR DESCRIPTION
**Pull request by mp-12301**
- Added Vite for bundling and local development.
- Added tasks for lint, dev, build and preview.

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -5,4 +5,6 @@ extends: standard-with-typescript
 parserOptions:
   ecmaVersion: latest
   sourceType: module
-rules: {}
+rules: {
+  import/no-absolute-path: off
+}
```
Comment (slimkrazy) [inline .eslintrc.yml]: Please add a trailing line break.

File: .eslintrc.yml [hunk 2]
```diff
@@ -5,4 +5,6 @@ extends: standard-with-typescript
 parserOptions:
   ecmaVersion: latest
```
Comment (slimkrazy) [inline .eslintrc.yml]: Don't we want to be explicit here? Especially considering we have `ES2020` in typescript config under `lib`?

**Review by slimkrazy — COMMENTED** [review]
Question about eslint config, but outside of that, LGTM.

File: .eslintrc.yml (hunk 2 above)
Comment (mp-12301) [reply .eslintrc.yml]: Correct, we have to use `ES2020` when we lint the code as well.

File: .github/workflows/pr-merged.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    types: [closed]
+jobs:
+  test:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
```
Comment (marian-r) [inline .github/workflows/pr-merged.yml]:
```suggestion
          node-version: 20
```

File: .github/workflows/pr-opened.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
```
Comment (marian-r) [inline .github/workflows/pr-opened.yml]:
```suggestion
          node-version: 20
```

**Review by marian-r — DISMISSED** [review]
Thanks

**Review by marian-r — APPROVED** [review]

**Pull request merged**